### PR TITLE
Expand LoadLibrary search paths

### PR DESCRIPTION
**Pull request by schellap**
@gkhanna79 PTAL.

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2144)

### CODE REVIEW

**Comment by schellap** [comment]
Fixes https://github.com/dotnet/corefx/issues/7357

File: src/corehost/common/pal.windows.cpp [hunk 1]
```diff
@@ -85,7 +85,7 @@ bool pal::getcwd(pal::string_t* recv)
 
 bool pal::load_library(const char_t* path, dll_t* dll)
 {
-    *dll = ::LoadLibraryW(path);
+    *dll = ::LoadLibraryExW(path, NULL, LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR | LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
```
Comment (gkhanna79) [inline src/corehost/common/pal.windows.cpp]: Please add a comment for this change. Otherwise, LGTM.

**Comment by schellap** [comment]
Fixes https://github.com/dotnet/cli/issues/2038

@glennc @troydai

**Comment by troydai** [comment]
@schellap can you give us some detail about how this would fix the #2038?

**Comment by schellap** [comment]
The reason corehost in sdk folder worked was because dotnet.dll was a standalone app with the api-forwarders. LoadLibrary would pick up from own directories fine.

Once you used the muxer LoadLibrary couldn't find the api forwarders. In your case, an UI would've popped up saying api-... blah was missing. We couldn't get this important piece of diagnostic info in that min kernel. The fix expands the search path to use default search paths, and module load directory as well.

**Comment by troydai** [comment]
Thanks. So what is it different in Antares from normal Windows environment. Because you theory only explains that the error message doesn't show up in Anates, which of course make the diagnosis harder but shouldn't the assembly searching logic the same?

**Comment by schellap** [comment]
This never reproed on Win 10 boxes because they don't require api-forwarders. You require api-forwarders for min clients.

**Comment by troydai** [comment]
Got it. Is possible to add test to automate this scenario on a platform other than windows 10?

**Pull request merged**